### PR TITLE
[FEATURE] Allow account+ledger lookup without auth

### LINKED ISSUE #105: Enable ledger to be looked up from account identifier
Right now when doing pathfinding we need to pass `source_account`, `source_ledger` and `source_username`, even though `source_account` combines the two other pieces of information. We're supposed to treat URIs as opaque so we can't assume that we can parse the username or ledger from the `source_account`.

My proposal would be to allow anyone (without authentication) to `GET` the account URI and return an object that contains the `ledger` and `account`.

One thing that would enable would be that your "account" that you share with people could actually be a service not tied to your ledger that would return your up to date payment info. If the standard behavior was for people to `GET` that endpoint first, you'd be able to update what that points to whenever you want to switch where you hold your money. This would be very useful for cases when you want to embed payment information in some data that can't be easily updated when you move accounts.

The first potential problem with this is that that endpoint would become a target for hackers because if they could change where that points to they could redirect where your money goes.

The second potential problem is that you might not want anyone to be able to find out if an account exists. That could be addressed by adding functionality to the ledgers to enable account holders to create aliases for their account that they could give to specific parties.

### PR DESCRIPTION
**Pull request by sentientwaffle**
Fixes #105 

Depends on https://github.com/interledger/five-bells-shared/pull/45

### CODE REVIEW

File: test/accountSpec.js [hunk 1]
```diff
@@ -106,27 +110,38 @@ describe('Accounts', function () {
         .end()
     })
 
-    it('should return 401 when not authenticated', function *() {
+    it('should return 200/404 when not authenticated, with partial data', function *() {
```
Comment (MatthewPhinney) [inline test/accountSpec.js]: Maybe this could be two tests? One for the existing account and one for the nonexistent one?

**Comment by sentientwaffle** [comment]
Ok, tests are split.

**Comment by clark800** [comment]
LGTM

**Comment by MatthewPhinney** [comment]
LGTM

**Pull request merged**